### PR TITLE
[Wasm] Fix support for TextBox clear button and header focus

### LINKED ISSUE #2132: [Wasm] TextBox clear button is not functional
## Current behavior

The XAML Clear button is not working.

## Expected behavior

The XAML Clear button is working

## How to reproduce it (as minimally and precisely as possible)

- Navigate to https://playgroundcanary.z19.web.core.windows.net/#textbox 
- Type some text in a textbox, try clearing the content

## Environment

https://dev.azure.com/uno-platform/Uno%20Platform/_build/results?buildId=9829&view=results

Nuget Package: 2.0.512-dev.4042

Package Version(s):

Affected platform(s):

- [ ] iOS
- [ ] Android
- [x] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

Visual Studio:

- [ ] 2017 (version: )
- [ ] 2019 (version: )
- [ ] for Mac (version: )

Relevant plugins:

- [ ] Resharper (version: )

### PR DESCRIPTION
**Pull request by jeromelaban**
GitHub Issue (If applicable): Fixes #2132 

## PR Type

What kind of change does this PR introduce?
- Bugfix

## What is the current behavior?

Tapping on the TextBox clear button on Wasm fails to clear the content of the TextBox.

## What is the new behavior?

The original behavior is restored for Wasm, though tapping twice on the button is required. See https://github.com/unoplatform/uno/issues/2138 for more information.

Tests have been added to validated the clear button behavior.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tested code with current [supported SDKs](https://github.com/unoplatform/uno/blob/master/README.md#uno-features)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [x] [Wasm UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences. Validate PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [ ] ~~Updated the [Release Notes]~~(https://github.com/unoplatform/uno/tree/master/doc/ReleaseNotes)
- [x] Associated with an issue (GitHub or internal)

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.wasm.cs [hunk 1]
```diff
@@ -27,18 +27,13 @@ public partial class TextBox : Control
 
 		partial void OnFocusStateChangedPartial(FocusState focusState)
```
Comment (davidjohnoliver) [inline src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.wasm.cs]: So... does this mean that programmatic focus won't work? (Assuming it worked before)

Comment (jeromelaban) [reply src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.wasm.cs]: It may not no. Focus management on Wasm is not advanced enough.

Comment (dr1rrb) [inline src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.wasm.cs]: ou can completely remove this partial method :)
```suggestion

```

File: src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.wasm.cs [hunk 2]
```diff
@@ -27,18 +27,13 @@ public partial class TextBox : Control
...
 		{
```
Comment (dr1rrb) [inline src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.wasm.cs]:
```suggestion
```

File: src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.wasm.cs [hunk 3]
```diff
@@ -27,18 +27,13 @@ public partial class TextBox : Control
...
-			if(focusState != FocusState.Unfocused)
-			{
-				FocusTextView();
-			}
 		}
```
Comment (dr1rrb) [inline src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.wasm.cs]:
```suggestion
```

File: src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.wasm.cs [hunk 4]
```diff
@@ -27,18 +27,13 @@ public partial class TextBox : Control
...
 
 		partial void OnAcceptsReturnChangedPartial(DependencyPropertyChangedEventArgs e)
```
Comment (dr1rrb) [inline src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.wasm.cs]: Same for this one
```suggestion
```

File: src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.wasm.cs [hunk 5]
```diff
@@ -27,18 +27,13 @@ public partial class TextBox : Control
...
 		{
```
Comment (dr1rrb) [inline src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.wasm.cs]:
```suggestion
```

File: src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.wasm.cs [hunk 6]
```diff
@@ -27,18 +27,13 @@ public partial class TextBox : Control
...
 
 		}
```
Comment (dr1rrb) [inline src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.wasm.cs]:
```suggestion
```

**Pull request merged**